RELEASES.md

```diff
@@ -5,7 +5,8 @@ Language
 --------
 - [make `const_err` show up in future breakage reports](https://github.com/rust-lang/rust/pull/97743/)
 - [allow unions with mutable references and tuples of allowed types](https://github.com/rust-lang/rust/pull/97995/)
-- [do not mark interior mutable shared refs as dereferenceable](https://github.com/rust-lang/rust/pull/98017/)
+- It is now considered valid to deallocate memory pointed to by a shared reference `&T` [if every byte in `T` is inside an `UnsafeCell`](https://github.com/rust-lang/rust/pull/98017/)
+- Unused tuple struct fields are now warned against in an allow-by-default lint, [`unused_tuple_struct_fields`](https://github.com/rust-lang/rust/pull/95977/). This lint will become warn-by-default in the future.
```

Comment:
This duplicates the text in the second bullet under "compiler"; could you consolidate? I like your text better, though I think I'd add something like "similar to the existing warning for other unused struct fields".